transform/log_address_index.go

```diff
@@ -127,21 +119,21 @@ func (i *LogAddressIndex) matchingBlocks(addrs []eth.Address, eventSigs []eth.Ha
 		}
 		sigsBitmap.Or(i.eventSigs[sigString])
 	}
-	if addrBitmap.IsEmpty() {
+
+	if len(addrs) == 0 { // ONLY try to match signatures
 		out := sigsBitmap.ToArray()
 		if len(out) == 0 {
 			return nil
 		}
 		return out
 	}
 
-	addrBitmap.And(sigsBitmap) // transforms addrBitmap
+	addrBitmap.And(sigsBitmap) // addrBitmap is changed in-place, becomes merged data
 	out := addrBitmap.ToArray()
 	if len(out) == 0 {
 		return nil
 	}
 	return out
```

Comment:
Why not return the roaring bitmap? I think it'll be way faster to query for membership containment than looping through a list of uints. It's designed for that.

Reply:
Before returning, you can also run Optimize on the bitmap, so it shakes it a bit, and optimizes its internal layout for quick querying. See: https://pkg.go.dev/github.com/RoaringBitmap/roaring/roaring64#Bitmap.RunOptimize

Reply:
RunOptimize is for further compression, it's not for query performance ???!

Reply:
si t'as une trâlée de 1,2,3,4,5,6,7, ça le compresse à range(1,7), ça se check plus vite que de looper les éléments on query time aussi